### PR TITLE
Doctests for some IO functions

### PR DESCRIPTION
**Pull request by kshyatt**

### CODE REVIEW

**Comment by fredrikekre** [comment]
`JuliaLang` -> `Julia` (?)

**Comment by StefanKarpinski** [comment]
Or change the sentence to "JuliaLang is a GitHub organization".

**Comment by kshyatt** [comment]
I changed the sentence and added a bunch more examples for `IOBuffer`

**Comment by iblis17** [comment]
current CI failure is https://github.com/JuliaLang/julia/pull/24500#issuecomment-349384270

File: base/io.jl [hunk 1]
```diff
@@ -84,7 +84,20 @@ Read a single value of type `T` from `stream`, in canonical binary representatio
 
     read(stream::IO, String)
 
-Read the entirety of `stream`, as a String.
+Read the entirety of `stream`, as a `String`.
```
Comment (fredrikekre) [inline base/io.jl]: I like when the function signature and examples match, perhaps change `` `stream` `` to `` `io` `` here and in the two signatures? Seems like `io` is used in almost all signatures.

File: base/io.jl [hunk 2]
```diff
@@ -99,6 +112,23 @@ You can write multiple values with the same `write` call. i.e. the following are
 
     write(stream, x, y...)
     write(stream, x) + write(stream, y...)
+
```
Comment (fredrikekre) [inline base/io.jl]: Same here I suppose.

File: base/io.jl [hunk 3]
```diff
@@ -170,6 +200,14 @@ wait_close(io::AbstractPipe) = (wait_close(pipe_writer(io)); wait_close(pipe_rea
     nb_available(stream)
 
 Return the number of bytes available for reading before a read from this stream or buffer will block.
+
```
Comment (fredrikekre) [inline base/io.jl]: And here.

File: base/iobuffer.jl [hunk 4]
```diff
@@ -262,6 +312,17 @@ isopen(io::GenericIOBuffer) = io.readable || io.writable || io.seekable || nb_av
 
 Obtain the contents of an `IOBuffer` as an array, without copying. Afterwards, the
 `IOBuffer` is reset to its initial state.
+
+# Examples
+```jldoctest
+julia> io = IOBuffer();
+
+julia> write(io, "JuliaLang is a GitHub organization.", "It has many members.")
```
Comment (fredrikekre) [inline base/iobuffer.jl]: `"It has ...` -> `" It has ...`

File: base/io.jl [hunk 5]
```diff
@@ -329,6 +367,19 @@ htol(x)
     isreadonly(stream) -> Bool
 
 Determine whether a stream is read-only.
+
+# Examples
```
Comment (fredrikekre) [inline base/io.jl]: And here; Perhaps I was completely wrong when I thought that most signatures used `io` :smile: But I still think it is nice to be consistent :slightly_smiling_face:

**Review by fredrikekre — APPROVED** [review]

**Comment by StefanKarpinski** [comment]
`io` is the most common name for an `IO` object at this point. However, Jeff was fond of using `s` for `IO` objects for a while which was fine except when there were also strings involved, which are conventionally called `s` as well – you can see the problem. So please use `io::IO` for new code and docs. We should at some points make a pass through everything and use more consistent argument names, but fortunately since positional argument names are insignificant by design, that's a 1.x job.

**Comment by kshyatt** [comment]
good to go?

**Pull request merged**